### ISSUE TITLE
Training with AMP (Mixed Precision) leads to incorrect (scaled) gradients being logged to Weights&Biases

### ISSUE THREAD

**Issue by konstantinjdobler** [PyTorchLightning/pytorch-lightning#11407]
Using `logger = WandbLogger(...)`  and `logger.watch(model, gradients="True")`, when setting `precision=16` in the `Trainer`, the gradient values logged to W&B seem to be the **scaled** gradients (scaled through the Automatic Mixed Precision). This is very confusing, as this makes it seem like e.g. gradient clipping is not working (because the scaled gradients can get very large) even though it is indeed working.

For example, I thought gradient clipping through setting `gradient_clip_val` in the `Trainer` wasn't working because the logged gradients were much larger than the clipping threshold. After spending more than one hour trying to figure out why `gradient_clip_val` isn't working, I figured out that it is working and the logged gradients are just scaled.
### Expected behavior
I believe the *unscaled* gradients should be logged. However I'm not sure if this issue arises because Pytorch Lightning does special stuff to make the gradient clipping easy or if W&B just doesn't work correctly for logging gradients in a  mixed precision setting.

### Environment
* CUDA:
	- GPU:
		- NVIDIA A100-SXM-80GB
		- NVIDIA A100-SXM-80GB
	- available:         True
	- version:           11.0
* Packages:
	- numpy:             1.22.0
	- pyTorch_debug:     False
	- pyTorch_version:   1.10.0
	- pytorch-lightning: 1.5.8
	- tqdm:              4.62.3
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- ELF
	- processor:         x86_64
	- python:            3.9.9
	- version:           # 91-Ubuntu SMP Thu Jul 15 19:09:17 UTC 2021

**Comment by rohitgr7** [comment]
I'd suggest using `Trainer(track_grad_norm=1)`. It logs gradients after they are unscaled. wandb might be logging them once they are available, but can't say how it works internally there.
